torch_xla/csrc/tensor.cpp

```diff
@@ -228,17 +228,23 @@ XLATensor::ShardingSpecPtr XLATensor::sharding_spec() const {
   return nullptr;
 }
 
-void XLATensor::SetShardingSpec(const ShardingSpec& sharding_spec) {
-  XLA_CHECK(GetIrValue().node != nullptr) << "Tyring to access a null cursor";
-  dynamic_cast<XlaNode*>(GetIrValue().node.get())
-      ->SetSharding(sharding_spec.sharding);
+void XLATensor::SetShardingSpec(const ShardingSpec& sharding) {
+  // Existing annotation must be cleared explicitly. We do not clear and
+  // overwrite the existing sharding on the user's behalf. This is a no-op if
+  // the same sharding already applied.
+  if (sharding_spec() == nullptr ||
```

Comment:
shouldn't we throw an error or log something if we call `SetShardingSpec` on a tensor with different sharding?

Reply:
silently no-op seems dangerous.